src/main/java/edu/kis/powp/jobs2d/features/CommandFactory.java

```diff
@@ -0,0 +1,83 @@
+package edu.kis.powp.jobs2d.features;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.gui.CommandManagerWindow;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+import edu.kis.powp.observer.Subscriber;
+
+public class CommandFactory implements Subscriber {
+    public Map<String, DriverCommand> commandsMap = new HashMap<>();
+    private DriverCommandManager driverCommandManager;
+    private CommandManagerWindow window;
```

Comment:
There are too many dependencies and responsibilities in this class. Try to modify the code to extract dependencies to GUI.